The docs about using pyodide on node.js are wrong

## 📚 Documentation

In particular, I'm referring to these instructions: https://pyodide.org/en/stable/usage/index.html#node-js 

To be honest, I don't know whether they are actually wrong or just unclear, but from the point of view of someone with Python experience but 0 knowledge of the JS world/tools, I didn't manage to run pyodide under node.

The doc says to `npm install pyodide`:
```
homer tmp $ mkdir /tmp/pyodide-experiment
homer tmp $ cd /tmp/pyodide-experiment/
homer pyodide-experiment $ npm install pyodide
npm WARN saveError ENOENT: no such file or directory, open '/tmp/pyodide-experiment/package.json'
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN enoent ENOENT: no such file or directory, open '/tmp/pyodide-experiment/package.json'
npm WARN pyodide-experiment No description
npm WARN pyodide-experiment No repository field.
npm WARN pyodide-experiment No README data
npm WARN pyodide-experiment No license field.

+ pyodide@0.20.0
added 6 packages from 5 contributors and audited 6 packages in 2.928s
found 0 vulnerabilities
```

Then it says "Download and extract Pyodide packages from [GitHub releases](https://github.com/pyodide/pyodide/releases)", but it doesn't specify *where* to download. I assume it's in the current directory?

```
homer pyodide-experiment $ wget -q https://github.com/pyodide/pyodide/releases/download/0.20.0/pyodide-build-0.20.0.tar.bz2
homer pyodide-experiment $ tar xf pyodide-build-0.20.0.tar.bz2 
homer pyodide-experiment $ ls -ld pyodide
drwxr-xr-x 3 antocuni antocuni 12288 apr 10 00:10 pyodide/
```

Then I tried to copy&paste the example inside a REPL:
```
homer pyodide-experiment $ node --experimental-repl-await
Welcome to Node.js v12.18.1.
Type ".help" for more information.
> let pyodide_pkg = await import("pyodide/pyodide.js");
(node:1035940) ExperimentalWarning: The ESM module loader is experimental.
Uncaught:
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/tmp/pyodide-experiment/node_modules/pyodide/pyodide.js' imported from /tmp/pyodide-experiment/repl
    at finalizeResolution (internal/modules/esm/resolve.js:284:11)
    at packageResolve (internal/modules/esm/resolve.js:615:14)
    at moduleResolve (internal/modules/esm/resolve.js:662:14)
    at Loader.defaultResolve [as _resolve] (internal/modules/esm/resolve.js:755:11)
    at Loader.resolve (internal/modules/esm/loader.js:97:40)
    at Loader.getModuleJob (internal/modules/esm/loader.js:243:28)
    at Loader.import (internal/modules/esm/loader.js:178:28)
    at importModuleDynamically (repl.js:373:45)
    at importModuleDynamicallyWrapper (internal/vm/module.js:432:21)
    at exports.importModuleDynamicallyCallback (internal/process/esm_loader.js:37:14) {
  code: 'ERR_MODULE_NOT_FOUND'
}
```

Then I tried to move the `pyodide` directory which was in the tarball inside `node_modules/pyodide`, but no luck:

```
homer pyodide-experiment $ mv pyodide node_modules/pyodide/
homer pyodide-experiment $ node --experimental-repl-await
> let pyodide_pkg = await import("pyodide/pyodide.js");
[...]
  code: 'ERR_MODULE_NOT_FOUND'
```

Then I tried to move the *content* of the pyodide directory into `node_modules/pyodide`:
```
homer pyodide-experiment $ mv node_modules/pyodide/pyodide/* node_modules/pyodide/
homer pyodide-experiment $ node --experimental-repl-await
Welcome to Node.js v12.18.1.
Type ".help" for more information.
> let pyodide_pkg = await import("pyodide/pyodide.js");
(node:1036517) ExperimentalWarning: The ESM module loader is experimental.
undefined
```

PROGRESS! But then it fails shortly after:
```
> let pyodide = await pyodide_pkg.loadPyodide();
Uncaught TypeError: pyodide_pkg.loadPyodide is not a function
    at repl:1:50
> pyodide_pkg
[Module] {  }
```

Then I gave up.

Still doesn't work:
```
homer x $ npm install pyodide
npm WARN saveError ENOENT: no such file or directory, open '/tmp/pyodide-experiment/package.json'
npm WARN enoent ENOENT: no such file or directory, open '/tmp/pyodide-experiment/package.json'
npm WARN pyodide-experiment No description
npm WARN pyodide-experiment No repository field.
npm WARN pyodide-experiment No README data
npm WARN pyodide-experiment No license field.

+ pyodide@0.20.0
updated 1 package and audited 9 packages in 1.248s
found 0 vulnerabilities

homer x $ node
Welcome to Node.js v12.18.1.
Type ".help" for more information.
> const { loadPyodide } = require("pyodide")//
Uncaught Error: Cannot find module 'pyodide'
Require stack:
- <repl>
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:966:15)
    at Function.Module._load (internal/modules/cjs/loader.js:842:27)
    at Module.require (internal/modules/cjs/loader.js:1026:19)
    at require (internal/modules/cjs/helpers.js:72:18) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ '<repl>' ]
}
> 
```

--- a/docs/usage/index.md
+++ b/docs/usage/index.md
@@ -81,14 +81,14 @@ To avoid this situation, one solution is to run {ref}`Pyodide in a WebWorker <us
 
 ```{note}
 The following instructions have been tested with Node.js 18.5.0. To use
-Pyodide with older versions of node, you might need special command line
-instructions, see below.
+Pyodide with older versions of Node, you might need to use  additional command line
+arguments, see below.
 ```
 
 It is now possible to install the
 [Pyodide npm package](https://www.npmjs.com/package/pyodide) in Node.js. To
-follow these instructions you need at least pyodide 0.21.0, which at the
-moment of writing has not been released yet. You can explicity ask npm to use
+follow these instructions you need at least Pyodide 0.21.0.
+You can explicitly ask npm to use
 the alpha version:
 
 ```
@@ -137,13 +137,13 @@ undefined
 2
 ```
 
-### Older versions of Node.js
+### Node.js versions <0.17
 
 - `Node.js` versions 14.x and 16.x: to use certain features of Pyodide you
   need to manually install `node-fetch`, e.g. by doing `npm install node-fetch`.
 
 - `Node.js v14.x`: you need to pass the option `--experimental-wasm-bigint`
-  when starting node. Note that this flag is not documented by `node --help`
+  when starting Node. Note that this flag is not documented by `node --help`
   and moreover, if you pass `--experimental-wasm-bigint` to node >14 it is an
   error:
 

--- a/docs/usage/index.md
+++ b/docs/usage/index.md
@@ -79,30 +79,88 @@ To avoid this situation, one solution is to run {ref}`Pyodide in a WebWorker <us
 
 ## Node.js
 
-As of version 0.18.0 Pyodide can experimentally run in Node.js.
-
-Install the [Pyodide npm package](https://www.npmjs.com/package/pyodide),
-
-```
-npm install pyodide
+```{note}
+The following instructions have been tested with Node.js 18.5.0. To use
+Pyodide with older versions of node, you might need special command line
+instructions, see below.
 ```
 
-Download and extract Pyodide packages from [GitHub
-releases](https://github.com/pyodide/pyodide/releases)
-(**pyodide-build-\*.tar.bz2** file). The version of the release needs to match
-exactly the version of this package.
+It is now possible to install the
+[Pyodide npm package](https://www.npmjs.com/package/pyodide) in Node.js. To
+follow these instructions you need at least pyodide 0.21.0, which at the
+moment of writing has not been released yet. You can explicity ask npm to use
+the alpha version:
 
-Then you can load Pyodide in Node.js as follows,
+```
+$ npm install "pyodide@>=0.21.0-alpha.2"
+```
+
+Once installed, you can run the following simple script:
 
 ```js
-let pyodide_pkg = await import("pyodide/pyodide.js");
-let pyodide = await pyodide_pkg.loadPyodide();
-await pyodide.runPythonAsync("1+1");
+// hello_python.js
+const { loadPyodide } = require("pyodide");
+
+async function hello_python() {
+    let pyodide = await loadPyodide();
+    return pyodide.runPythonAsync("1+1");
+}
+
+hello_python().then(result => {
+    console.log("Python says that 1+1 =", result);
+})
 ```
 
-```{note}
-To start the Node.js REPL with support for top level await, use `node --experimental-repl-await`.
 ```
+$ node hello_python.js
+Loading distutils
+Loaded distutils
+Python initialization complete
+Python says that 1+1= 2
+```
+
+Or you can use the REPL. To start the Node.js REPL with support for top level
+await, use `node --experimental-repl-await`:
+
+```
+$ node --experimental-repl-await
+Welcome to Node.js v18.5.0.
+Type ".help" for more information.
+> const { loadPyodide } = require("pyodide");
+undefined
+> let pyodide = await loadPyodide();
+Loading distutils
+Loaded distutils
+Python initialization complete
+undefined
+> await pyodide.runPythonAsync("1+1");
+2
+```
+
+### Older versions of Node.js
+
+- `Node.js` versions 14.x and 16.x: to use certain features of Pyodide you
+  need to manually install `node-fetch`, e.g. by doing `npm install
+  node-fetch`.
+
+- `Node.js v14.x`: you need to pass the option `--experimental-wasm-bigint`
+  when starting node.  Note that this flag is not documented by `node --help`
+  and moreover, if you pass `--experimental-wasm-bigint` to node >14 it is an
+  error:
+
+```
+$ node -v
+v14.20.0
+
+$ node --experimental-wasm-bigint hello_python.js
+warning: no blob constructor, cannot create blobs with mimetypes
+warning: no BlobBuilder
+Loading distutils
+Loaded distutils
+Python initialization complete
+Python says that 1+1= 2
+```
+
 
 ```{eval-rst}
 .. toctree::

--- a/docs/usage/index.md
+++ b/docs/usage/index.md
@@ -102,13 +102,13 @@ Once installed, you can run the following simple script:
 const { loadPyodide } = require("pyodide");
 
 async function hello_python() {
-    let pyodide = await loadPyodide();
-    return pyodide.runPythonAsync("1+1");
+  let pyodide = await loadPyodide();
+  return pyodide.runPythonAsync("1+1");
 }
 
-hello_python().then(result => {
-    console.log("Python says that 1+1 =", result);
-})
+hello_python().then((result) => {
+  console.log("Python says that 1+1 =", result);
+});
 ```
 
 ```
@@ -140,11 +140,10 @@ undefined
 ### Older versions of Node.js
 
 - `Node.js` versions 14.x and 16.x: to use certain features of Pyodide you
-  need to manually install `node-fetch`, e.g. by doing `npm install
-  node-fetch`.
+  need to manually install `node-fetch`, e.g. by doing `npm install node-fetch`.
 
 - `Node.js v14.x`: you need to pass the option `--experimental-wasm-bigint`
-  when starting node.  Note that this flag is not documented by `node --help`
+  when starting node. Note that this flag is not documented by `node --help`
   and moreover, if you pass `--experimental-wasm-bigint` to node >14 it is an
   error:
 
@@ -160,7 +159,6 @@ Loaded distutils
 Python initialization complete
 Python says that 1+1= 2
 ```
-
 
 ```{eval-rst}
 .. toctree::

